### PR TITLE
Reorder React

### PR DESCRIPTION
**Pull request by stevebrownlee**
After discussion with @jisie about her journey through teaching React, the chapters were reordered to delay the teaching of dynamic routing by a day or two.

### CODE REVIEW

File: book-4-the-apprentice/chapters/FUNCTIONS_AS_PROPS.md [hunk 1]
```diff
@@ -113,7 +104,7 @@ render() {
 }
 ```
 
-So just like a primitive value like a string or an integer, you can pass function references from a parent component to a child component. The child component can then specify when that functionality should be invoked, even though it was defined on its parent.
+So just like a primitive value, such as a string or an integer, you can pass function references from a parent component to a child component. The child component can then specify when that functionality should be invoked, even though it was defined on its parent.
 
 ## Practice: Remove Customers
```
Comment (Rian501) [inline book-4-the-apprentice/chapters/FUNCTIONS_AS_PROPS.md]: I don't think we have customers in the kennel DB exercise... should it be employees?

Comment (stevebrownlee) [reply book-4-the-apprentice/chapters/FUNCTIONS_AS_PROPS.md]: @Rian501 Good catch. I'll change it to employees. Also added another practice exercise that has the students making a CustomerList and a Customer component, which should also implement the functionality of deleting.

**Comment by stevebrownlee** [comment]
@brendalong I made the first few chapters have the definition of the class and the export statement on separate lines. Then I show them combined and explain it. From that point on, they will see them combined.

**Comment by brendalong** [comment]
The re-order looks fine.

**Comment by stevebrownlee** [comment]
Ok, I've received approval from a quorum of the instruction team. I'm merging this branch now since I'm about to embark upon a new refactor of book 2 based on conversations with @brendalong @megducharme @JoeShep and @jisie

**Pull request merged**